File: apps/ios/Loonext/Features/Tasks/MemberPickerSheet.swift
import SwiftUI

/// Member picker: a sheet with a search field over the active members
/// (GET /v1/members), a "(you)" marker, and an optional Unassigned entry.
/// Callers own the fetch — the sheet is pure UI, mirroring the Android
/// MemberPickerSheet 1:1.
@MainActor
struct MemberPickerSheet: View {
    let members: [Member]
    let meUserId: String
    let selectedUserId: String?
    let showUnassigned: Bool
    let onPick: @MainActor (_ userId: String?) -> Void

    @State private var query = ""
    @Environment(\.dismiss) private var dismiss
    @Environment(\.appLocale) private var appLocale

    private var matches: [Member] {
        let active = members.filter { $0.deactivated_at == nil }
        let trimmed = query.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return active }
        return active.filter { $0.display_name.localizedCaseInsensitiveContains(trimmed) }
    }

    var body: some View {
        VStack(spacing: 0) {
            HStack(spacing: 8) {
                Image(systemName: "magnifyingglass")
                    .font(.scaled(14, weight: .medium))
                    .foregroundStyle(BrandColor.muted400)
                TextField(
                    AppStrings.translate(appLocale, "contactsTasks.searchTeammates"),
                    text: $query
                )
                    .font(.golos(13))
                    .textInputAutocapitalization(.never)
                    .autocorrectionDisabled()
            }
            .padding(.horizontal, 15)
            .padding(.vertical, 10)
            .background(BrandColor.paper, in: Capsule())
            .padding(.horizontal, 16)
            .padding(.vertical, 12)

            List {
                if showUnassigned && query.isBlank {
                    pickerRow(
                        name: AppStrings.translate(appLocale, "contactsTasks.unassigned"),
                        avatarName: nil,
                        selected: selectedUserId == nil
                    ) {
                        onPick(nil)
                        dismiss()
                    }
                }
                ForEach(matches, id: \.user_id) { member in
                    let base = member.display_name.isBlank
                        ? AppStrings.translate(appLocale, "contactsTasks.teammate")
                        : member.display_name
                    let youSuffix = AppStrings.translate(
                        appLocale, "contactsTasks.youSuffix"
                    )
                    pickerRow(
                        name: member.user_id == meUserId ? base + youSuffix : base,
                        avatarName: member.display_name.isBlank ? nil : member.display_name,
                        selected: selectedUserId == member.user_id
                    ) {
                        onPick(member.user_id)
                        dismiss()
                    }
                }
                if matches.isEmpty {
                    Text(AppStrings.translate(appLocale, "contactsTasks.noTeammatesMatch"))
                        .font(.golos(13))
                        .foregroundStyle(BrandColor.muted500)
                        .listRowSeparator(.hidden)
                }
            }
            .listStyle(.plain)
            .scrollContentBackground(.hidden)
        }
        .background(BrandColor.canvas.ignoresSafeArea())
        .presentationDetents([.medium, .large])
    }

    private func pickerRow(
        name: String,
        avatarName: String?,
        selected: Bool,
        onTap: @escaping @MainActor () -> Void
    ) -> some View {
        HStack(spacing: 12) {
            if let avatarName {
                InitialsAvatar(name: avatarName, size: 32)
            } else {
                Color.clear.frame(width: 32, height: 32)
            }
            Text(name)
                .font(.golos(13.5, weight: .semibold))
                .foregroundStyle(BrandColor.ink)
            Spacer()
            if selected {
                Image(systemName: "checkmark")
                    .foregroundStyle(BrandColor.olive)
                    .accessibilityLabel(
                        AppStrings.translate(appLocale, "contactsTasks.selected")
                    )
            }
        }
        .contentShape(Rectangle())
        .onTapGesture(perform: onTap)
    }
}

// MARK: - Previews

#Preview("Member picker") {
    MemberPickerSheet(
        members: [
            Member(
                id: "mb1",
                user_id: "u1",
                role: MemberRole.owner,
                deactivated_at: nil,
                created_at: "2026-06-01T12:00:00Z",
                display_name: "Sam Carpenter"
            ),
            Member(
                id: "mb2",
                user_id: "u2",
                role: MemberRole.member,
                deactivated_at: nil,
                created_at: "2026-06-02T12:00:00Z",
                display_name: "Alex Mason"
            ),
            Member(
                id: "mb3",
                user_id: "u3",
                role: MemberRole.member,
                deactivated_at: "2026-07-01T12:00:00Z", // filtered out (deactivated)
                created_at: "2026-06-03T12:00:00Z",
                display_name: "Former Teammate"
            ),
        ],
        meUserId: "u1",
        selectedUserId: "u2",
        showUnassigned: true,
        onPick: { _ in }
    )
}

File: apps/android/app/src/main/kotlin/com/loonext/android/features/tasks/MemberPickerSheet.kt
package com.loonext.android.features.tasks

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.loonext.android.core.model.Member
import com.loonext.android.ui.common.InitialsAvatar

/**
 * cmdk-style member picker: a bottom sheet with a search field over the
 * active members (GET /v1/members), a "(you)" marker, and an optional
 * Unassigned entry. Callers own the fetch — the sheet is pure UI.
 */
@Composable
fun MemberPickerSheet(
    members: List<Member>,
    meUserId: String,
    selectedUserId: String?,
    showUnassigned: Boolean,
    onPick: (userId: String?) -> Unit,
    onDismiss: () -> Unit,
) {
    var query by remember { mutableStateOf("") }
    val active = members.filter { it.deactivated_at == null }
    val matches = if (query.isBlank()) active
    else active.filter { it.display_name.contains(query.trim(), ignoreCase = true) }

    ModalBottomSheet(onDismissRequest = onDismiss) {
        Column(Modifier.fillMaxWidth()) {
            OutlinedTextField(
                value = query,
                onValueChange = { query = it },
                label = { Text("Search teammates") },
                singleLine = true,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp, vertical = 8.dp),
            )
            LazyColumn(Modifier.fillMaxWidth()) {
                if (showUnassigned && query.isBlank()) {
                    item(key = "unassigned") {
                        PickerRow(
                            name = "Unassigned",
                            avatarName = null,
                            selected = selectedUserId == null,
                            onClick = { onPick(null) },
                        )
                    }
                }
                items(matches, key = { it.user_id }) { member ->
                    PickerRow(
                        name = member.display_name.ifBlank { "Teammate" } +
                            if (member.user_id == meUserId) " (you)" else "",
                        avatarName = member.display_name.ifBlank { null },
                        selected = selectedUserId == member.user_id,
                        onClick = { onPick(member.user_id) },
                    )
                }
                if (matches.isEmpty()) {
                    item(key = "empty") {
                        Text(
                            "No teammates match.",
                            style = MaterialTheme.typography.bodyMedium,
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                            modifier = Modifier.padding(16.dp),
                        )
                    }
                }
                item(key = "bottom-space") { Spacer(Modifier.height(24.dp)) }
            }
        }
    }
}

@Composable
private fun PickerRow(
    name: String,
    avatarName: String?,
    selected: Boolean,
    onClick: () -> Unit,
) {
    Row(
        Modifier
            .fillMaxWidth()
            .clickable(onClick = onClick)
            .padding(horizontal = 16.dp, vertical = 12.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        if (avatarName != null) {
            InitialsAvatar(avatarName, size = 32.dp)
        } else {
            Box(Modifier.width(32.dp))
        }
        Spacer(Modifier.width(12.dp))
        Text(
            name,
            style = MaterialTheme.typography.bodyLarge,
            modifier = Modifier.weight(1f),
        )
        if (selected) {
            Icon(
                Icons.Filled.Check,
                contentDescription = "Selected",
                tint = MaterialTheme.colorScheme.primary,
            )
        }
    }
    HorizontalDivider(color = MaterialTheme.colorScheme.outlineVariant)
}
